src/Task/File/Write.php

```diff
@@ -143,6 +143,20 @@ public function regexReplace($pattern, $replacement)
         return $this;
     }
 
+    /**
+     * Append the provided text to the end of the buffer if the
+     * provided regex pattern does not match any of the text already
+     * in the buffer.
+     *
+     * @param string $pattern
+     * @param string
+     */
+    public function textIfMatch($pattern, $text, $shouldMatch = true)
```

Comment:
hello again, its bad naming!
if you need a 3 lines of text to describe the method behavior it's something wrong here. The name of a method should be self-describing here, so my suggestions to have 2 methods:

* appendIfMatches($pattern, $text)
* appendUnlessMatches($pattern, $text)

..smth like that